python/selfie-lib/tests/PerCharacterEscaper_test.py

```diff
@@ -0,0 +1,63 @@
+import pytest
+
+from selfie_lib import PerCharacterEscaper
+
+
+class TestPerCharacterEscaper:
+    def test_performance_optimization_self(self):
+        escaper = PerCharacterEscaper.self_escape("`123")
+        abc = "abc"
+        # Using 'is' to check for the exact same object might not behave as in Kotlin, use == for equality in Python
+        assert escaper.escape(abc) == abc
+        assert escaper.unescape(abc) == abc
```

Comment:
In the test this is ported from, note `assertSame` vs `assertEquals`

https://github.com/diffplug/selfie-python-wip/blob/5e984657f4d14707bc09c5d8bb804059963b398c/jvm/selfie-lib/src/commonTest/kotlin/com/diffplug/selfie/PerCharacterEscaperTest.kt#L29-L37

In Python, `assertSame` becomes `is`, and `assertEquals` becomes `==`.